### PR TITLE
updated for csproj/net 2.0

### LINKED ISSUE #459: how to write compatible applications
Hi,
  I'm looking for informations on how to write web applications and libraries that could be easily switched between .net core and .net framework.
  The problem I see that in this first period, until most the 3rd party libraries are compatible with .net core, could be possibile I have to add libraries that are not ready for core, so I can solve or moving all the application to .net framework or move the code to another web app using .net framework and sharing for example the data model already wrote, but at this point I need to be able to compile a lib both for core and .net fx, but I haven't found any documentation about this.

### PR DESCRIPTION
**Pull request by mairaw**
I've noticed this topic had still references to project.json and didn't contain the latest TFMs.

Related to #2091 

[Internal Preview URL](https://review.docs.microsoft.com/en-us/dotnet/core/porting/third-party-deps?branch=pr-en-us-2860)

Fixes #459 

edit by @billwagner Added "fixes" comment based on @mairaw comment on that issue.

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
Looks good, though I definitely wouldn't merge this without a :shipit: from @terrajobst

**Comment by mairaw** [comment]
Ok, I'll wait for @terrajobst's review! Thanks @cartermp!

File: docs/core/porting/third-party-deps.md
Comment (terrajobst) [inline docs/core/porting/third-party-deps.md]: It seems a bit weird to mention Windows here because it makes it sound we're only talking about how you can port to .NET Core running on Windows. We really mean that we assume the customer is *starting* on Windows with .NET Framework. The end result might run on Linux.

File: docs/core/porting/third-party-deps.md
Comment (terrajobst) [inline docs/core/porting/third-party-deps.md]: While true, we need to call out the .NET Framework compatibility mode we have now. See my upcoming blog post on .NET Standard 2.0. It has a detailed walkthrough on how this works. In essence, it allows people to consume .NET Framework libraries when targeting .NET Standard 2.0 (or any .NET implementation of it). They get a warning telling them that library might not work, but they are no longer blocked as they are today. After testing, they can suppress the warning.

**Review by terrajobst — CHANGES_REQUESTED** [review]
We need to call out the new .NET Framework compatibility mode that comes with .NET Standard 2.0.

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [inline docs/core/porting/third-party-deps.md]: I saw that. Will add.

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [inline docs/core/porting/third-party-deps.md]: Makes sense. I'll change it to make it more clear.

**Comment by mairaw** [comment]
Sorry for the super mega delay on addressing the feedback here @terrajobst. Please take a look.

I'm also wondering if we wanna keep the portable TFMs and the pre-release TFMs in this topic. Thoughts?

File: docs/core/porting/third-party-deps.md
Comment (svick) [inline docs/core/porting/third-party-deps.md]: > The easiest way to inspect NuGet Package folders is to use the NuGet Package Explorer tool. After downloading it, use the following steps to see the folder names:
> …
> Alternatively, you can see the TFMs that each package supports on nuget.org under the **Dependencies** section of the package page.

How is installing a Windows-only tool and following a 5-step process easier than checking a website? Shouldn't it be the other way around, i.e. talk about nuget.org first and mention NPE as the alternative?

File: docs/core/porting/third-party-deps.md
Comment (svick) [inline docs/core/porting/third-party-deps.md]: Nit: the primary way of distributing NPE is the Windows Store, which means you don't really download it, you install it. So I would say:

> After installing it, use the following steps to see the folder names:

File: docs/core/porting/third-party-deps.md
Comment (svick) [inline docs/core/porting/third-party-deps.md]: How do I do that? Would a link to some documentation about this (possibly https://docs.microsoft.com/nuget/schema/msbuild-targets#packagetargetfallback) be useful here?

File: docs/core/porting/third-party-deps.md
Comment (svick) [inline docs/core/porting/third-party-deps.md]: I think "on Visual Studio" and "on your favorite code editor" should use "in" instead of "on".

File: docs/core/porting/third-party-deps.md
Comment (svick) [inline docs/core/porting/third-party-deps.md]: Should this link directly to the section about NuGet packages (https://docs.microsoft.com/visualstudio/ide/how-to-suppress-compiler-warnings#suppressing-warnings-for-nuget-packages)?

Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: Yes! That section didn't exist when I added this link. 😄

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: That's a good suggestion @svick, while we still don't have our own MSBuild reference topic for .NET Core.

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: I think you're right. Prepositions are my Achilles heel. @rpetrusha to confirm. 😄

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: I think it's a good idea. I think the issue is that NuGet doesn't always have that info. So instead of easiest, we could say that the tool is the more reliable way to check or something. See the package we use here as an example: https://www.nuget.org/packages/Huitian.PowerCollections/ and you'll see the dependencies are not listed

**Comment by mairaw** [comment]
@rpetrusha @BillWagner or @JRAlexander can one of you also do an UE review? Thanks! 

I think this one should be ready to go now (finally).

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: These could be links to the relevant sections. I think that that would enhance the readability of the topic.
also, most --> the most

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: a few ways --> two ways

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: Using --> Using the

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: on --> with

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: no comma after So

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: (TFM) --> (TFMs)

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: no paragraph break.

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: on --> in

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: "At least..." shouldn't be a separate sentence.

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: "next" is best eliminated.

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: filesystem --> filesystem
Add a parenthetical description of what the APIPort tool does.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
You've done a good job with the revisions, @mairaw. These are mostly nits, with a few other comments.

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: There's no paragraph break here (I'd need an extra line in Markdown to create that, so keeping as-is)

image

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: I think you mean to change it to file system?

**Review by mairaw — COMMENTED** [review]
Thanks for your thorough review @rpetrusha. I appreciate it!

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: nit: the **Dependencies** --> **Dependencies**

File: docs/core/porting/third-party-deps.md
Comment (rpetrusha) [inline docs/core/porting/third-party-deps.md]: nit: no comma after So

**Review by rpetrusha — APPROVED** [review]
Three nits, @mairaw, You can decide whether you want to address them before you merge.

File: docs/core/porting/third-party-deps.md
Comment (BillWagner) [inline docs/core/porting/third-party-deps.md]: The last two sentences in this paragraph are really important for customers to understand. Is there a way to make them stand out more? Make them their own paragraph, or highlight them in some other way?

**Review by BillWagner — APPROVED** [review]
This looks great @mairaw   

I only had one question, and I don't have a great way to solve it, so :shipit: when ready.

File: docs/core/porting/third-party-deps.md
Comment (mairaw) [reply docs/core/porting/third-party-deps.md]: Let me know what you think of the changes @BillWagner

**Comment by mairaw** [comment]
I've addressed all the remaining feedback. @terrajobst please take a look to see if now you can approve the PR. 

My final question is: do we still wanna list PCL and pre-release TFMs in this topic? 

Thanks!

**Comment by terrajobst** [comment]
> My final question is: do we still wanna list PCL and pre-release TFMs in this topic?

I think it's topical here as it's all about compat with the past when migrating.

**Review by terrajobst — APPROVED** [review]

**Comment by mairaw** [comment]
Cool. Thanks @terrajobst. 🎆

**Pull request merged**
